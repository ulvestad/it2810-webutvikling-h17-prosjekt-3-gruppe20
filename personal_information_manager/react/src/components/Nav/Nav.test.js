import React from 'react';
import ReactDOM from 'react-dom';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import { expect } from 'chai';
import { shallow, mount } from 'enzyme';
import { Link } from 'react-router-dom';
import {BrowserRouter} from 'react-router-dom'

import Nav from './Nav';

const jestExpect = global.expect;

beforeEach(() => {
  localStorage.getItem.reset();
  localStorage.setItem.reset();
});

it('renders without crashing', () => {
  shallow(<Nav />);
});

it('renders correctly', () => {
  const navWrappedInRouter = (
      <BrowserRouter>
        <Nav/>
      </BrowserRouter>
  );
  const tree = renderer.create(navWrappedInRouter).toJSON();
  jestExpect(tree).toMatchSnapshot();
});

it('runs updateNotificationBadge when clicked', () => {
  const update = sinon.spy(Nav.prototype, 'updateNotificationBadge');
  const wrapper = shallow(<Nav />);

  wrapper
    .find(Link)
    .first()
    .simulate('click');
  expect(update).to.have.property('calledOnce', true);
});
