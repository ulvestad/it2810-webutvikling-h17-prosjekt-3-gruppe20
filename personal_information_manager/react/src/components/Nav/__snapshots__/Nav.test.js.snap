// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
  id="mainNav"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      id="navtext"
      onClick={[Function]}
    >
      <img
        alt="logo"
        id="navlogo"
        src="/img/logo.png"
      />
      Personal Information Manager 
    </a>
    <a
      className="navbar-brand"
      href="/"
      id="navtextToogle"
      onClick={[Function]}
    >
      <img
        alt="logo"
        id="navlogoToogle"
        src="/img/logo.png"
      />
    </a>
    <button
      aria-controls="navbarResponsive"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarResponsive"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarResponsive"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item nav-link"
        >
          <a
            href="/calender"
            onClick={[Function]}
          >
            Calender
          </a>
        </li>
        <li
          className="nav-item nav-link"
        >
          <a
            href="/todo"
            onClick={[Function]}
          >
            Todo
          </a>
        </li>
        <li
          className="nav-item nav-link"
        >
          <a
            href="/notes"
            onClick={[Function]}
          >
            Notes
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
